### PR TITLE
Add API whitelist

### PR DESCRIPTION
**Pull request by viossat**
# Description

Allow specified addresses in `config.json` to access all API commands.
Accepts both IP and CIDR. `127.0.0.1` and `::1` are hard-coded as whitelisted.

To accept all IP:
```json
{
  "api": {
    "whitelistedAddresses": ["0.0.0.0/0", "::/0"]
  }
}
```

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

# How Has This Been Tested?

I tried to access permitted and protected API commands from whitelisted and unwhitelisted addresses.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have tested my code extensively
- [x] I have selected the `develop` branch as the target branch

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
       "storeTransactions",
       "getTrytes"
     ],
+    "whitelistedAddresses": [],
```
Comment (nuriel77) [inline config.json]: Afaik, current behavior is to whitelist `::1` and `127.0.0.1` by default, maybe this should be specified here by default, keeps it more visible, avoid confusion?

File: plugins/webapi/plugin.go [hunk 2]
```diff
@@ -75,6 +78,19 @@ func configure(plugin *node.Plugin) {
 		}
 	}
 
+	// Load whitelisted addresses
+	whitelist := append([]string{"127.0.0.1", "::1"}, parameter.NodeConfig.GetStringSlice("api.whitelistedAddresses")...)
+	if len(whitelist) > 0 {
```
Comment (IoTMOD) [inline plugins/webapi/plugin.go]: `len(whitelist)` is always 2 here, so you don't need this check

File: plugins/webapi/plugin.go [hunk 3]
```diff
@@ -75,6 +78,19 @@ func configure(plugin *node.Plugin) {
...
+		for _, entry := range whitelist {
+			_, ipnet, err := cnet.ParseCIDROrIP(entry)
+			if err != nil {
+				log.Warnf("Invalid whitelist address: %s", entry)
+			} else {
+				whitelistedNetworks = append(whitelistedNetworks, ipnet.IPNet)
+			}
```
Comment (IoTMOD) [inline plugins/webapi/plugin.go]: We can remove the `else` here
```suggestion
			if err != nil {
				log.Warnf("Invalid whitelist address: %s", entry)
				continue
			} 
		        whitelistedNetworks = append(whitelistedNetworks, ipnet.IPNet)
```

**Review by IoTMOD — CHANGES_REQUESTED** [review]
Thanks for your PR!

File: plugins/webapi/plugin.go (hunk 2 above)
Comment (viossat) [reply plugins/webapi/plugin.go]: True, I added the two addresses after the condition

**Review by IoTMOD — APPROVED** [review]

**Pull request merged**